test/e2e/serial/tests/workload_placement_tmpol.go

```diff
@@ -1568,3 +1540,21 @@ func isHugePageInAvailable(rl corev1.ResourceList) bool {
 	}
 	return false
 }
+
```

Comment:
I think this is the right call.
It's about time we do some factoring in this area, the argument list was a bit wild already.
I think this is the right improvement we need, a code extraction without (yet) a redesign of the flow. We are grouping a bunch of functions together for practicality, there's not yet a separate data type (with state whatsoever).

My only gripe is naming. "tmPolicyToolbox" feels clunky. I'd go with something like
```
- perPolicyHelpers
- tmPolicyFuncs
- perTMPolicyFuncs
```
Probably "tmPolicyFuncs" is the best (= less ugly) I can think atm.
I'd also rename the member funcs like
```
type tmPolicyFuncs struct {
	policyName()            func() string
	setupPadding            func(fxt *e2efixture.Fixture, nrtList nrtv1alpha1.NodeResourceTopologyList, padInfo paddingInfo) []*corev1.Pod
	checkConsumedResources  func(nrtInitial, nrtUpdated nrtv1alpha1.NodeResourceTopology, required corev1.ResourceList) (string, error)
	filterMatchingResources func(nrts []nrtv1alpha1.NodeResourceTopology, requests corev1.ResourceList) []nrtv1alpha1.NodeResourceTopology
}
```

the idea is that every member of `tmPolicyFuncs` should be a `func` (and not a data field). The creation funcs can look like:

```
func newPodScopeTMPolicyFuncs() tmPolicyFuncs {
	return tmPolicyToolbox{
		policyName:              func() string { return  nrtv1alpha1.SingleNUMANodePodLevel },
		setupPadding:            setupPaddingPodLevel,
		checkConsumedRes:        e2enrt.CheckZoneConsumedResourcesAtLeast,
		filterMatchingResources: e2enrt.FilterAnyZoneMatchingResources,
	}
}
```

Reply:
The naming is always something I'm struggling with.
But the most important thing is that you liked the idea :) 
I'll fix the naming asap.
Thanks for the review.